### PR TITLE
tcp buf config

### PR DESCRIPTION
**Pull request by olofkallander**
Small TCP send buffer is filled when RC is used and stream became corrupted.
Changed TcpEndpoint to maintain packet structure in stream.

### CODE REVIEW

File: transport/TcpEndpoint.cpp [hunk 1]
```diff
@@ -277,21 +268,54 @@ void TcpEndpoint::continueSend()
...
+        if (bytesSent < _remainder.getLength())
         {
-            logger::warn("failed send to %s, err (%d) %s",
-                _name.c_str(),
-                _peerPort.toString().c_str(),
-                rc,
-                _socket.explain(rc));
+            memory::Packet tmp;
+            tmp.append(reinterpret_cast<uint8_t*>(_remainder.get()) + bytesSent, _remainder.getLength() - bytesSent);
+            tmp.copyTo(_remainder);
```
Comment (RicardoMDomingues) [inline transport/TcpEndpoint.cpp:287]: `std::memmove` would avoid temporary allocation and double copy

File: transport/TcpEndpoint.cpp [hunk 2]
```diff
@@ -277,21 +268,54 @@ void TcpEndpoint::continueSend()
...
+        }
+        else
+        {
+            _remainder.setLength(0);
         }
+        if (_remainder.getLength() > 0)
```
Comment (RicardoMDomingues) [inline transport/TcpEndpoint.cpp:293]: isn't this `if` the same of line `283`( `if (bytesSent < _remainder.getLength())`)? Maybe you can lof the `warn` and `return` on the first `if`

File: transport/TcpEndpoint.cpp [hunk 3]
```diff
@@ -277,21 +268,54 @@ void TcpEndpoint::continueSend()
...
+    }
+    else if (bytesSent < packet.getLength() + sizeof(shim))
+    {
+        bytesSent -= sizeof(uint16_t);
+        _remainder.append(packet.get() + bytesSent, packet.getLength() - bytesSent);
+        logger::debug("partial packet sent %zu / %zu, err %d", _name.c_str(), bytesSent, packet.getLength(), rc);
+    }
+    else if (bytesSent == 0)
+    {
+        logger::warn("discarding packet. err %d", _name.c_str(), rc);
```
Comment (RicardoMDomingues) [inline transport/TcpEndpoint.cpp:318]: Never runs because is caught by 1st `if`

File: transport/RtcSocket.cpp [hunk 4]
```diff
@@ -243,25 +241,24 @@ int RtcSocket::sendAggregate(const struct iovec* messages, uint16_t messageCount
     {
         rc = ::sendmsg(_fd, &header, MSG_DONTWAIT);
         errorCode = errno;
-        if (rc == totalLength || errorCode == EINPROGRESS)
+        if (rc <= 0 && (errorCode == EAGAIN || errorCode == EWOULDBLOCK))
```
Comment (RicardoMDomingues) [inline transport/RtcSocket.cpp:244]: `errno` is not set when `rc` is zero

**Review by RicardoMDomingues — APPROVED** [review]

**Comment by olofkallander** [comment]
RTC-12722

**Review by RicardoMDomingues — APPROVED** [review]

**Pull request merged**